website/src/parsers/js/transformers/babel-macros/babel-plugin-macros.js

```diff
@@ -0,0 +1,263 @@
+/** 
+ *  Just for the proof of concept.
+ * 
+ *  I had to modifiy the BablePluginMacro 
+ *  to be able to inject a custom require function 
+ *  that would always resolve to the same macro.
+ * 
+ *  No idea if this could be worked around with webpack...
+ */
+
+
+const p = require('path')
+// const printAST = require('ast-pretty-print')
+
+const macrosRegex = /[./]macro(\.js)?$/
+
+// https://stackoverflow.com/a/32749533/971592
+class MacroError extends Error {
+  constructor(message) {
+    super(message)
+    this.name = 'MacroError'
+    /* istanbul ignore else */
+    if (typeof Error.captureStackTrace === 'function') {
+      Error.captureStackTrace(this, this.constructor)
+    } else if (!this.stack) {
+      this.stack = new Error(message).stack
+    }
+  }
+}
+
+function createMacro(macro, options = {}) {
+  if (options.configName === 'options') {
+    throw new Error(
+      `You cannot use the configName "options". It is reserved for babel-plugin-macros.`,
+    )
+  }
+  macroWrapper.isBabelMacro = true
+  macroWrapper.options = options
+  return macroWrapper
+
+  function macroWrapper(args) {
+    const {source, isBabelMacrosCall} = args
+    if (!isBabelMacrosCall) {
+      throw new MacroError(
+        `The macro you imported from "${source}" is being executed outside the context of compilation with babel-plugin-macros. ` +
+          `This indicates that you don't have the babel plugin "babel-plugin-macros" configured correctly. ` +
+          `Please see the documentation for how to configure babel-plugin-macros properly: ` +
+          'https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/user.md',
+      )
+    }
+    return macro(args)
+  }
+}
+
+function macrosPlugin(babel, { require: _require }) {
+
+  function interopRequire(path) {
+    // eslint-disable-next-line import/no-dynamic-require
+    const o = _require ? _require(path) : require(path);
```

Comment:
Optional use the passed in require to overwrite the default behaviour of `require`